src/ReleaseHistory.md

```diff
@@ -5,6 +5,7 @@
 * BREAKING: Update `IAnalysisContext.LogToolNotification` method to add `ReportingDescriptor` parameter. This is required in order to populated `AssociatedRule` data in `Notification` instances. The new method has an option value of null for the `associatedRule` parameter to maximize build compatibility. [#2604](https://github.com/microsoft/sarif-sdk/pull/2604)
 * BREAKING: Correct casing of `LogMissingreportingConfiguration` helper to `LogMissingReportingConfiguration`. [#2599](https://github.com/microsoft/sarif-sdk/pull/2599)
 * BREAKING: Change type of `MaxFileSizeInKilobytes` from int to long in `IAnalysisContext` and other classes.  [#2599](https://github.com/microsoft/sarif-sdk/pull/2599)
+* BREAKING: Properties changed from `int`(32bit) to `BigInteger`(unlimited): `Address.AbsoluteAddress`, `Address.RelativeAddress`, `Address.Length`, `Address.OffsetFromParent`, `Artifact.Offset`, `Artifact.Length`, `Invocation.ExitCode`, `Invocation.ExitSignalNumber`, `LocationRelationship.Target`, `Region.ByteOffset`, `Region.ByteLength`, `Result.OccurrenceCount`. [#2572](https://github.com/microsoft/sarif-sdk/pull/2572)
 * BREAKING: For `Guid` properties defined in SARIF spec, updated Json schema to use `uuid`, and updated C# object model to use `Guid?` instead of `string`. [#2555](https://github.com/microsoft/sarif-sdk/pull/2555)
```

Comment:
@shaopeng-gh Curious: are there any performance implications for switching from Int32 to BigInteger?

Reply:
This is good information. Allocating a StringBuilder instance is a very heavyweight operation at local code execution scope.

We should definitely test deserialization behavior slowdown. You'd likely benefit from testing a synthetic file, i.e., create a SARIF file that very intensively produces BigInteger properties (not a real world example).